tests/boot/test_mcuboot/swapped_app/Kconfig

```diff
@@ -0,0 +1,9 @@
+# Copyright 2022 NXP
+# SPDX-License-Identifier: Apache-2.0
+source "Kconfig.zephyr"
+# Workaround for not being able to have commas in macro arguments
+DT_CHOSEN_Z_CODE_PARTITION := zephyr,code-partition
+
+config BUILD_OUTPUT_ADJUST_LMA
```

Comment:
This does not work for devices where the secondary slot it not part of the primary flash, e.g. in an external flash.

Reply:
So such boards shouldn't be used as build targets.

Reply:
That seems like a big limitation. Would it not be better to have a Kconfig option to generate for the secondary slot instead? The list of devices might not be large at present but it is going to only grow with time (devices where slot1 and slot0 differ):
```
arm/usb_kw24d512/usb_kw24d512.dts
arm/rm1xx_dvk/rm1xx_dvk.dts
arm/pinnacle_100_dvk/pinnacle_100_dvk.dts
arm/lpcxpresso55s36/lpcxpresso55s36.dts
arm/lpcxpresso55s28/lpcxpresso55s28.dts
arm/lpcxpresso55s16/lpcxpresso55s16_common.dtsi
arm/lpcxpresso55s06/lpcxpresso55s06_common.dtsi
arm/lpcxpresso54114/lpcxpresso54114_m4.dts
arm/frdm_kw41z/frdm_kw41z.dts
arm/disco_l475_iot1/disco_l475_iot1.dts
arm/bt610/bt610.dts
arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts
```

Reply:
It seems that the second slot being on an external device isn't tested much because dev boards usually don't have external flash. But, it does seem common in production. I think it is an important case to consider.

Reply:
> It seems that the second slot being on an external device isn't tested much because dev boards usually don't have external flash. But, it does seem common in production. I think it is an important case to consider.

I had a quick grep through the boards and it seems 63 boards have QSPI entries in their boards files, 63/500 is 13% of all supported boards, which isn't a small number.

Reply:
To clarify, "isn't tested very much" by things like our testing infrastructure. I'm sure there are lots of targets that do this, and that it is pretty common in production.

Reply:
> This does not work for devices where the secondary slot is not part of the primary flash, e.g. in an external flash.

What would the required steps here be in order to generate an image for the secondary flash? The image needs to be build targeting the primary slot, then the programming tool (pyocd, jlink,...) needs to be instructed to load the image at a given offset, correct? Or if the image cannot be loaded using jlink or another west runner, what would be the method to flash it?

Perhaps the work in #50393 could be leveraged to pass a load address to the runner?

Reply:
For nordic parts, if the secondary slot it in QSPI then it is memory mapped, 0x12000000 for nRF52840 and 0x10000000 for nRF5340, this can be programmed directly to the QSPI flash using nrfjprog, I would assume other flash runners would not support programming the QSPI. There is offset given to nrfjprog with e.g. a bin file, it needs a properly addressed hex file. For SPI flash on other nordic parts, e.g. nRF52832, they cannot be flashed from nrfjprog, flashing of such a flash would be down to whomever creates the board (it would need to be a custom flash runner)